### ISSUE TITLE
Simplifying Layouts

### ISSUE THREAD

**Issue by vkbo** [vkbo/novelWriter#835]
## Current State

The current scheme of document layouts is slowly becoming redundant. The initial idea was to give the writer a lot more control of how each individual document was formatted on export or build, but it's time to simplify this. There are some areas where the layouts are useful, but almost all of the features they provide are now available in other ways:

* Unnumbered Chapters: Adding an asterisk in front of the chapter title provides the same feature, so this is redundant. A single way of indicating this would also be less confusing.
* Partition and Title Page: These layouts by default centre all text. With the release of 1.4, there is no longer a strict need to use layouts for this as this version will add syntax for aligning individual paragraphs. Level one headers should still be centred by default. These layouts also provide automated page breaks, but so does the generic layout type "Page".
* Book, Chapter and Scene: These are already all the same, and only serve a purpose of indicating to the user what the file is intended for. Since novelWriter now keeps track of the level of the first title in each file, there are no need for individual layouts for the same.
* Note: Anything that isn't in the Novel root folder is a note. That distinction is easy to detect and doesn't require a separate layout.
* Page: This is actually a useful distinction from files that are part of the novel itself as page breaks are treated differently around these.

## Proposed Changes

For the user side of things, the following changes can be made:

* Merge layouts "Partition" and "Title Page" into "Page".
* Merge layouts "Unnumbered", "Book", "Chapter" and "Scene" into "Story".
* Add a setting to mark a document as a "Standalone Page" and let everything else be "Story". This only applies to files in the "Novel" root folder.
* Add a simple tool or option to convert a note to a story file and vice versa. This will clear the status or importance flag.
* Replace the "Flags" column in the project tree with a "Status" column that only shows the item's status or importance.
* Add icons that indicate if a "Story" file has a chapter or scene heading as the first title. The icon can be the same as the current text document icon, but with a "C" or an "S" on them, for instance.
* Add a different icon for "Note" files. Perhaps something looking like a post-it note?

For the code side of things, the following changes can be made:

* Reduce the layout field to "NOTE", "PAGE" and "STORY".
* The project tree must read the header level of a file from the index and assign an icon.
* Simplify the assignment of item class to files. They should always be the same as the root folder they're under.

**Comment by jyhelle** [comment]
I agree with all of these. 
Conversion of notes to story files and back is one of my major activities, so I put a strong vote on this one ! (in fact it's rather part of a note to either a scene or chapter, someone told me I use the "snowflake" : two lines in a note become a chapter, that chapter grows until it is split into a dozen of scenes, etc)

**Comment by vkbo** [comment]
Yes, I use a variation of the snowflake method too. I don't use notes for it, I just do it directly in "Book" type files and use the split tool to split the scenes later.

In any case, the main purpose of these changes is that the distinction between the layouts is no longer needed for export. At least not in the files that become the body text  of the novel. All necessary information is extracted automatically. I'm still thinking of practical ways to indicate in the project tree which files are chapters and scenes though, but I think I will make different icons for them in the end.

I started implementing these changes last night and found that keeping the "Title Page" layout is handy, so I've reduced it to "Title", "Page", "Note" and everything else is "Story". I'm fairly happy with the changes, so I will probably keep them and merge them into the main code after I've released 1.4.
